crates/services/gas_price_service/src/v1/service.rs

```diff
@@ -158,31 +153,41 @@ where
         block_bytes: u64,
         block_fees: u64,
     ) -> anyhow::Result<()> {
-        let capacity = self.validate_block_gas_capacity(block_gas_capacity)?;
+        let capacity = Self::validate_block_gas_capacity(block_gas_capacity)?;
+        let mut storage_tx = self.storage_tx_provider.begin_transaction()?;
+
+        for da_block_costs in self.da_block_costs_buffer.drain(..) {
```

Comment:
In the case of an error, you will lose `da_block_costs_buffer` data. Maybe it is better just to iterate here and clear it later after you committed changes to the database.